commands/init.go

```diff
@@ -188,18 +188,58 @@ func loadGenesis(ctx context.Context, rep repo.Repo, sourceName string) (consens
 		return nil, err
 	}
 
-	if len(ch.Roots) != 1 {
-		return nil, fmt.Errorf("expected car with only a single root")
+	// need to check if we are being handed a car file with a single genesis block or an entire chain.
+	bsBlk, err := bs.Get(ch.Roots[0])
+	if err != nil {
+		return nil, err
 	}
-
-	gif := func(cst *hamt.CborIpldStore, bs blockstore.Blockstore) (*block.Block, error) {
-		var blk block.Block
-
-		if err := cst.Get(ctx, ch.Roots[0], &blk); err != nil {
+	cur, err := block.DecodeBlock(bsBlk.RawData())
+	if err != nil {
+		return nil, err
+	}
+	// the root block of the car file has parents, this file must contain a chain
+	for !cur.Parents.Equals(block.UndefTipSet.Key()) {
+		bsBlk, err := bs.Get(cur.Parents.ToSlice()[0])
+		if err != nil {
+			return nil, err
+		}
+		cur, err = block.DecodeBlock(bsBlk.RawData())
+		if err != nil {
 			return nil, err
 		}
+	}
```

Comment:
I'm not fond of this walking the whole file to find the genesis block business. A couple alternatives that come to mind:
- Store the genesis block as the first entry after the CAR file header.
- (Ab)use the CAR file header and put the genesis block cid in `ch.Roots[0]` and the head TipSets in `ch.Roots[1:]`
- Wrap the CAR file in something specifically made for this use case.

Other ideas?